tests/utils/search/test_gridsearch.py

```diff
@@ -70,4 +69,15 @@ def test_generate_grid(self):
         """Tests if generate_grid function returns expected value."""
         self.assertEqual(self.g_mini.generate_grid(),
                          [{'c1': 1, 'c2': 6, 'k': 5, 'w': 0.9, 'p': 0},
-                          {'c1': 2, 'c2': 6, 'k': 5, 'w': 0.9, 'p': 0}])
\ No newline at end of file
+                          {'c1': 2, 'c2': 6, 'k': 5, 'w': 0.9, 'p': 0}])
+
+class Instantiate(Base):
+
+    def test_optimizer_type_fail(self):
+        """Tests that :code:`optimizer` of type :code:`string` raises
+        :code:`TypeError`"""
+        bad_optimizer = 'LocalBestPSO'  # a string instead of a class object
+        with self.assertRaises(TypeError):
+            g = GridSearch(bad_optimizer, self.n_particles, self.dimensions,
```

Comment:
I see, I think I understand what you mean earlier. I guess this is the way to go when initializing objects.